apps/src/lib/node/ledger/storage/rocksdb.rs

```diff
@@ -302,6 +302,69 @@ impl RocksDB {
 
         println!("Done writing to {}", full_path.to_string_lossy());
     }
+
+    /// Rollback to previous block
+    pub fn rollback(&mut self) -> Result<()> {
+        let last_block = self.read_last_block()?.expect("No block was found");
+        let mut batch = WriteBatch::default();
+
+        // Revert the non-height-prepended metadata storage keys
+        for metadata_key in [
+            "next_epoch_min_start_height",
+            "next_epoch_min_start_time",
+            "tx_queue",
+        ] {
+            let key = Key::from(metadata_key.to_owned().to_db_key());
+            let previous_key =
+                Key::from(format!("pred/{}", metadata_key).to_db_key());
+            let previous_value = self.read_subspace_val(&previous_key)?.expect(
```

Comment:
Does this keep track of whether the key was changed in the last block; ie we need to make certain that we only revert to the `pred` key if it was actually different on block `n-1`

Reply:
As far as I understood these three keys are updated with every block